exchanges/okcoin/okcoin_ws_trade.go

```diff
@@ -0,0 +1,191 @@
+package okcoin
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"sync"
+
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+)
+
+var (
+	waitingSignatureLock sync.Mutex
+	waitingSignatures    = []string{}
```

Comment:
I'm not seeing any removal of signatures in code. This will cause an endlessly appending list which will get slower over time

Reply:
That's a good suggestion thanks. I will use map[string]struct{} for that.